File: src/main/java/info/skyblond/archivedag/model/entity/UserRoleEntity.java
package info.skyblond.archivedag.model.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "user_role")
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@IdClass(UserRoleEntity.class)
public class UserRoleEntity implements Serializable {
    @Id
    @Column(name = "username", updatable = false)
    String username;

    @Id
    @Column(name = "role_name", nullable = false)
    String role;

    public UserRoleEntity(String username, String role) {
        this.username = username;
        this.role = role;
    }
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arstue/entity/UserRoleEntity.kt
package info.skyblond.archivedag.arstue.entity

import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "user_role")
@IdClass(UserRoleEntity::class)
class UserRoleEntity(
    @Id
    @Column(name = "username", updatable = false)
    val username: String,

    @Id
    @Column(name = "role_name", nullable = false)
    val role: String
) : Serializable
